templates/admin/stacktrace.tmpl

```diff
@@ -18,6 +19,16 @@
 		</div>
 	</div>
 </div>
+<div class="ui small basic modal" id="download-stacktrace-modal">
+	<div class="ui icon header">
+		{{svg "octicon-download" 16 "close inside"}}
+		{{.locale.Tr "admin.monitor.stacktrace.download_stacktrace"}}
+	</div>
+	{{template "base/delete_modal_actions" .}}
+	<div class="hide" id="stacktrace-to-download">
```

Comment:
Why not just:

1. User click the download button
2. Download from a HTTP URL directly

?

Then no need that complex dialog nor the JS code

Reply:
No. Then they get a different stacktrace.

Reply:
I mean , could there be no stacktrace on the UI? Just some links to help to collect problems.

The complex UI doesn't help users. As an end user, they should just download and report the stacktrace file.

The end users could do nothing even if they see the stacktrace on the UI.....

Reply:
FYI, when I was using GitLab, they have an all-in-one tool. It collects everything into a file, and what I need to do is just using the tool and sending the file to them, then they can help to resolve problems.

Making the diagnosis too complex doesn't benefit end users IMO.

Reply:
> I mean , could there be no stacktrace on the UI? Just some links to help to collect problems.

I'm going to add a direct download stacktrace profile below to get the stacktraces without seeing them.

> The complex UI doesn't help users. As an end user, they should just download and report the stacktrace file.

The pretty stacktrace helps me and SEVERAL bugs have been solved using it. It has helped me a large number of times already.

> The end users could do nothing even if they see the stacktrace on the UI.....

Not every user is incapable and whilst the purpose of this UI is to help us to help users we should enable users and developers to help themselves. A pretty UI can be helpful for us to solve issues in a way that the opaque stacktrace format is not.

Reply:
> > I mean , could there be no stacktrace on the UI? Just some links to help to collect problems.
> 
> I'm going to add a direct download stacktrace profile below to get the stacktraces without seeing them.

If it means more complex, then it's not necessary IMO. My initial idea is about keeping the system simple but complete and useful.

While I do not think keeping the downloaded stacktrace file exactly the same as the UI list is meaningful -- everytime you refresh the page, you get a different stacktrace ..... Even if they are different, they are all helpful for resolving problem equally, and maybe the UI shown list is not helpful but the downloaded is helpful, everything is possible.

Reply:
Fundamentally it's really important to be able to download the stacktrace that you have seen. Given these are dynamic it's possible that you may actually only see the issue on the stacktrace that you have and if you try get another the issue will be gone.

We need the download. 

---
> FYI, when I was using GitLab, they have an all-in-one tool. It collects everything into a file, and what I need to do is just using the tool and sending the generated file to them, then they can help to resolve problems.
>
> Making the diagnosis system too complex doesn't benefit end users IMO.

If I knew what we generally needed I would do that - but in general we don't need cpu-profiles etc. 

We do need a way of reading the logs and I will get to that but ... let's get general routes in and then we can have pared back routes to get the general things. Right now we have no easy way of getting pprof profiles from users and that fundamentally limits our ability to use them for diagnostics. This PR makes that a hell of a lot easier.